project/src/main/puzzle/critter/spears.gd

```diff
@@ -568,6 +576,50 @@ func _shift_rows(bottom_y: int, direction: Vector2) -> void:
 		for spear_line in _rows_for_spear(spear):
 			spear_bottom_row = max(spear_line, spear_bottom_row)
 		spear.visible = spear_bottom_row >= PuzzleTileMap.FIRST_VISIBLE_ROW
+	
+	_detect_and_resolve_spear_conflicts()
+
+
+## Detects any spear overlaps, removing any small spears which overlap larger spears.
+##
+## Line clears do not remove spears which haven't popped out yet, so it's possible for two spears to end up in the
+## same row overlapping the same cells. In this unlikely scenario, we remove the smaller of the overlapping spears.
+func _detect_and_resolve_spear_conflicts() -> void:
+	# key: (Vector2) playfield cell which will be occupied by a spear when it is popped in
+	# value: (Spear) spear which is currently on the playfield
+	var spears_by_veg_cell: Dictionary = {}
+	
+	for spear in _veg_cells_by_spear.keys():
+		for i in range(_veg_cells_by_spear[spear].size()):
+			var veg_cell: Vector2 = _veg_cells_by_spear[spear][i]
+			var conflicting_spear: Spear = spears_by_veg_cell.get(veg_cell)
+			if not conflicting_spear:
+				spears_by_veg_cell[veg_cell] = spear
+				continue
+			
+			# Calculate which spear is larger; the larger spear wins the conflict.
+			var _spears_by_size := [spear, conflicting_spear]
+			_spears_by_size.sort_custom(self, "_compare_spears_by_size")
+			var large_spear: Spear = _spears_by_size[0]
+			var small_spear: Spear = _spears_by_size[1]
+			
+			# write the large spear to spears_by_veg_cell, and remove references to the small spear
+			spears_by_veg_cell[veg_cell] = large_spear
+			for zoom_cell in spears_by_veg_cell.keys():
```

Comment:
'zoom_cell'?

Reply:
Do we have to iterate over this entire map? Can't we iterate over the small spear's cells?

Reply:
Refactored logic and changed errant loop variable name.

`for small_cell in _veg_cells_by_spear[small_spear]:`